gui/create-item.lua

```diff
@@ -184,11 +211,183 @@ local function getCreatureRaceAndCaste(caste)
  return df.global.world.raws.creatures.list_creature[caste.index],df.global.world.raws.creatures.list_caste[caste.index]
 end
 
+function createCorpsePiece(creator, bodypart, partlayer, creatureID, casteID, generic, quality) -- this part was written by four rabbits in a trenchcoat (ppaawwll)
+ -- (partlayer is also used to determine the material if we're spawning a "generic" body part (i'm just lazy lol))
+ quality = math.max(0, math.min(5, quality - 1))
+ local creat = tonumber(creatureID)
+ -- get the actual raws of the target creature
+ local creatorRaceRaw = df.global.world.raws.creatures.all[creat]
+ local cast = tonumber(casteID)
+ local bodpart = tonumber(bodypart)
+ local patlayer = tonumber(partlayer)
```

Comment:
Similar suggestions for all of these. In general, creating similar variables with deliberate typos/missing letters to distinguish them is bound to cause confusion, and in this case, it looks like you don't really want the original values (before tonumber()) anyway, so there's no point in keeping them around.

Reply:
true.